src/kv/untyped_map_handle.cpp

```diff
@@ -206,7 +206,9 @@ namespace kv::untyped
     // std::lower_bound()/std::upper_bound() and loop over it, interleaves
     // with the local writes.
 
-    if (from == to || to < from)
+    if (
+      from.has_value() && to.has_value() &&
+      (from.value() == to.value() || to.value() < from.value()))
```

Comment:
Not sure why the condition was like this in the first place, but I think this line should be equivalent to `from.value() >= to.value() `? Also, what if `from = std::nullopt` but `to` has a value?

Reply:
Fixing the condition, not sure why it was like that. If `from = std::nullopt` and `to` has a value then this is an open ended range so we iterate from the first value up to (and excluding) `to`.

Reply:
I don't think the condition bit will want to work nicely with `>=` so I'll leave it for another PR. Works fine currently still anyway.